### PR TITLE
Implement Observer to DriverManager

### PR DESCRIPTION
**Pull request by mario0000o7**

### CODE REVIEW

**Comment by kowallus** [comment]
Please resolve conflicts

File: src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java [hunk 1]
```diff
@@ -11,12 +11,12 @@
...
 
 import edu.kis.powp.appbase.gui.WindowComponent;
-import edu.kis.powp.jobs2d.command.manager.DriverCommandManager;
+import edu.kis.powp.jobs2d.command.manager.CommandManager;
 import edu.kis.powp.observer.Subscriber;
 
 public class CommandManagerWindow extends JFrame implements WindowComponent {
 
-    private DriverCommandManager commandManager;
+    private CommandManager commandManager;
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java]: Refactor (renaming) could be in the separate PR and it sure shouldn't be in the same commit with adding an Observer. 
Please revert renaming.

File: src/main/java/edu/kis/powp/jobs2d/features/DriverFeature.java [hunk 2]
```diff
@@ -4,8 +4,9 @@
 import edu.kis.powp.jobs2d.Job2dDriver;
 import edu.kis.powp.jobs2d.drivers.DriverManager;
 import edu.kis.powp.jobs2d.drivers.SelectDriverMenuOptionListener;
+import edu.kis.powp.observer.Subscriber;
 
-public class DriverFeature {
+public class DriverFeature implements Subscriber {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/features/DriverFeature.java]: Do not add new responsibilities to DriverFeature. Please implement a separate Subscriber class.

**Review by kowallus — CHANGES_REQUESTED** [review]
Poor approach to git.
Please fix listed issues.

File: src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java (hunk 1 above)
Comment (tobiasz135) [reply src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java]: Name has been already changed by someone else.

**Review by tobiasz135 — COMMENTED** [review]
Task problems have been fixed

File: src/main/java/edu/kis/powp/jobs2d/events/DriverFeatureSubscriber.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package edu.kis.powp.jobs2d.events;
+
+import edu.kis.powp.jobs2d.features.DriverFeature;
+import edu.kis.powp.observer.Subscriber;
+
+public class DriverFeatureSubscriber implements Subscriber {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/events/DriverFeatureSubscriber.java]: All subscribers of DriverManager can be considered DriverFeatureSubscriber.
The name of the subscribe should indicate what is done upon notification (update method logic).

File: src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java [hunk 4]
```diff
@@ -75,11 +75,7 @@ private static void setupDrivers(Application application) {
 
         driver = new RecordingDriver(new LineDriverAdapter(drawerController, LineFactory.getSpecialLine(), "special"));
         DriverFeature.addDriver("Special line Simulator", driver);
-
         DriverFeature.addDriver("Logger + line driver", composite);
-
-
-        DriverFeature.updateDriverInfo();
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java:82]: I am not sure if the correct name of a driver appears on startup now. Please verify.

**Review by kowallus — CHANGES_REQUESTED** [review]
Fix name of the observer class

File: src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java (hunk 4 above)
Comment (tobiasz135) [reply src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java:82]: The name appears correctly except for "Line + logger driver" due to lack of toString() function in that driver.

File: src/main/java/edu/kis/powp/jobs2d/events/DriverFeatureSubscriber.java (hunk 3 above)
Comment (tobiasz135) [reply src/main/java/edu/kis/powp/jobs2d/events/DriverFeatureSubscriber.java]: Name updated

**Review by tobiasz135 — COMMENTED** [review]
Task issues have been fixed.

**Review by kowallus — APPROVED** [review]
OK

**Pull request merged**